packages/hebao_v2/contracts/base/WalletFactory.sol

```diff
@@ -96,20 +98,21 @@ contract WalletFactory is WalletDeployer
             config.inheritor,
             config.feeRecipient,
             config.feeToken,
-            config.feeAmount
+            feeAmount
         );
 
         emit WalletCreated(wallet, config.owner);
     }
 
     function _validateRequest(
         WalletConfig calldata config,
-        uint                  salt
+        uint feeAmount
         )
         private
         view
     {
         require(config.owner != address(0), "INVALID_OWNER");
+        require(feeAmount <= config.maxFeeAmount, "INVALID_FEE_AMOUNT");
```

Comment:
Please move this check out of this function so we can remove  the`feeAmount` parameter.